### ISSUE TITLE
Vim mode should filter output

### ISSUE THREAD

**Issue by jonsterling** [RedPRL/redtt#235]
I feel that the vim mode should perhaps filter output and *only* display things that are in the standard message format:

```
filename:span [tag]:
  indented output, followed by at least an extra line of whitespace


```

It may be, for instance, that redtt eventually dumps output that is meant to instruct the editor mode to do something specific, rather than be displayed to the user. What do you think?

**Comment by favonia** [comment]
We should design a proper API or protocol instead of relying on a fragile convension. Maybe an interactive mode for plug-ins. (Or just the language server?) It is very possible that some non-standard messages happen to have the "correct" format that would confuse the filter.

**Comment by jonsterling** [comment]
@favonia I agree! I propose we study and possibly adopt the Language Server Protocol. This has the benefit of being supported across a broad collection of editors, and is becoming a standard.

**Comment by favonia** [comment]
I looked at the protocol and was disappointed by the choice of JSON and its lack of formal specification (to our standard). JSON might be understandable from a convenience point of view, but repeating the same mistakes made by HTTP 1.x is unacceptable. However, this might still be our best choice.

**Comment by jonsterling** [comment]
@favonia Yeah... :( Well, if you have other ideas, they are very welcome too
